_rules/inline-link-in-paragraph-is-distinguishable-be4d0c.md

```diff
@@ -0,0 +1,475 @@
+		text-decoration: none;
+	}
+</style>
+<p>To read about WAI, <a class="test" href="http://w3.org/WAI">follow this link to the WAI webpage</a>.</p>
+```
+
+#### Passed Example 11
+
+This link, that is distinguishable by color (hue) from the other text in the same line, has an icon that appears when the link is hovered. This icon indicates the element is a link.
+
+```html
+<style>
+	p {
+		color: black;
+		background-color: white;
+	}
+	a {
+		color: blue;
+		background-color: white;
+		text-decoration: none;
+	}
+	.hide {
+		display: none;
+	}
+	a:hover img.hide {
+		display: inline;
+	}
+</style>
+<p>
+	Read about WAI on the
+	<a href="http://w3.org/WAI">WAI webpage <img class="hide" src="/test-assets/be4d0c/icon.png" alt="" /></a>.
+</p>
+```
+
+#### Passed Example 12
+
+This link had a descendant element that is distinguishable by color (hue) from the other text in the same line and has a distinguishable bottom border when it is hovered or focused.
+
+```html
+<style>
+	a {
+		text-decoration: none;
+	}
+
+	a:hover span,
+	a:focus span {
+		border-bottom: 2px solid currentColor;
+	}
+</style>
+
+<p>
+	Read about WAI on the <a href="http://w3.org/WAI"><span>WAI webpage</span></a
+	>.
+</p>
+```
+
+#### Passed Example 13
+
+This link is distinguishable by color (hue) from the other text in the same line in both link history states. The link has distinguishing color and background color. The color and background color are different when in the visited state, but these are also distinguishing colors.
+
+```html
+<style>
+	p {
+		color: black;
+		background-color: white;
+	}
+	a:link {
+		color: white;
+		background-color: blue;
+		text-decoration: none;
+	}
+	a:visited {
+		color: blue;
+		background-color: lightgray;
+	}
+</style>
+<p>Read about WAI on the <a href="http://w3.org/WAI">WAI webpage</a>.</p>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This link, that is distinguishable by color (hue) from the other text in the same line, has no other visual cues of being recognized as a link with the underline removed. Even though the link has a different color (blue) from the other text (black) the low difference in contrast between them means they are not distinguishing colors.
+
+```html
+<style>
+	p {
+		color: black;
+		background-color: white;
+	}
+	a.test {
+		color: blue;
+		background-color: white;
+		text-decoration: none;
+	}
+</style>
+<p>Read about WAI on the <a class="test" href="http://w3.org/WAI">WAI webpage</a>.</p>
+```
+
+#### Failed Example 2
+
+This link, that is distinguishable by color (hue) from the other text in the same line, has a different style because of its color and foreground color. But when the link has been visited, the style is the same of the other text on the same line.
+
+```html
+<style>
+	p {
+		color: black;
+		background-color: white;
+	}
+	a {
+		color: lightblue;
+		background-color: #111111;
+		text-decoration: none;
+	}
+	a:visited {
+		color: black;
+		background-color: white;
+	}
```

Comment:
Should it be kept with a `color:blue` so that it still has a different hue from the rest of the text 🤔 
If a user somehow arrives on the page with the link already visited, it wouldn't even be applicable due to the same hue…

So, I'm not sure that written that way it is a failure of 1.4.1… (in the `:visited` state, the author is not using color, so 1.4.1 does not apply).

Which might need an update of the rule 😱 Either taking the link history in Applicability ("different hue in each of the link history states"); or back in the expectation ("For all link history state with different hue"), but that raises the question of which state to test the Applicability in…
I guess the fully explicit way would be A: "different hue in at least one link history state", E: "For each link history state with different hue"…

Reply:
Nice catch... thinking about it, this is my understanding:
- 1.4.1 applies to links with link states where there are differences in hue, and only to those link states
- accordingly, the expectation should be checked only in the link states where 1.4.1 is applicable

So, I agree with your (fully explicit) suggestion :-)